src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -245,14 +413,19 @@ ss::future<result<void>> service::initiate_rpk_debug_bundle_collection(
 
     // Kick off the wait by waiting for the process to finish and then emplacing
     // the result
-    ssx::spawn_with_gate(_gate, [this]() {
-        return _rpk_process->_rpk_process->wait()
-          .then([this](auto res) { _rpk_process->_wait_result.emplace(res); })
-          .handle_exception_type([this](const std::exception& e) {
-              lg.error(
-                "wait() failed while running rpk debug bundle: {}", e.what());
-              _rpk_process->_wait_result.emplace(
-                ss::experimental::process::wait_exited{1});
+    ssx::spawn_with_gate(_gate, [this, job_id]() {
+        return _rpk_process->wait()
+          .then([this, job_id](auto) {
+              auto hold = _gate.hold();
```

Comment:
I'm not totally sure I get why you need to hold the gate here. is it because the background task completes when you return the `_process_control_mutex.get...` future below? 

I don't write enough continuation style code 😅